clang/lib/AST/DeclCXX.cpp

```diff
@@ -826,6 +842,14 @@ void CXXRecordDecl::addedMember(Decl *D) {
               ? !Constructor->isImplicit()
               : (Constructor->isUserProvided() || Constructor->isExplicit()))
         data().Aggregate = false;
+
+      // A trivially relocatable class is a class:
+      // -- where no eligible copy constructor, move constructor, copy
+      // assignment operator, move assignment operator, or destructor is
+      // user-provided,
+      if (Constructor->isUserProvided() && (Constructor->isCopyConstructor() ||
```

Comment:
This does not appear to do what the comment says -- `IsNaturallyTriviallyRelocatable` is cleared even when the constructor is ineligible. Maybe this should be moved to `addedEligibleSpecialMemberFunction`?

Reply:
Addressed in the latest commit. Thanks for bringing it up!